pkg/controller/publishingstrategy/publishingstrategy_controller.go

```diff
@@ -284,9 +426,72 @@ func (r *ReconcilePublishingStrategy) Reconcile(request reconcile.Request) (reco
 		}
 		return reconcile.Result{}, nil
 	}
+
 	return reconcile.Result{}, nil
 }
 
+// getIngressName takes the domain name and returns the first part
+func getIngressName(dnsName string) string {
+	firstPeriodIndex := strings.Index(dnsName, ".")
+	newIngressName := dnsName[:firstPeriodIndex]
+	return newIngressName
+}
+
+// newApplicationIngressControllerCR creates a new IngressController CR
+func newApplicationIngressControllerCR(ingressControllerCRName, scope, dnsName string, certificate *corev1.LocalObjectReference, matchLabels map[string]string) (*operatorv1.IngressController, error) {
+	loadBalancerScope := operatorv1.LoadBalancerScope("")
+	switch scope {
+	case "internal":
+		loadBalancerScope = operatorv1.InternalLoadBalancer
+	case "external":
+		loadBalancerScope = operatorv1.ExternalLoadBalancer
+	default:
+		return &operatorv1.IngressController{}, errors.New("ErrCreatingIngressController")
+	}
+
+	return &operatorv1.IngressController{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ingressControllerCRName,
+			Namespace: ingressControllerNamespace,
+		},
+		Spec: operatorv1.IngressControllerSpec{
+			DefaultCertificate: certificate,
+			Domain:             dnsName,
+			EndpointPublishingStrategy: &operatorv1.EndpointPublishingStrategy{
+				Type: operatorv1.LoadBalancerServiceStrategyType,
+				LoadBalancer: &operatorv1.LoadBalancerStrategy{
+					Scope: loadBalancerScope,
+				},
+			},
+			RouteSelector: &metav1.LabelSelector{
+				MatchLabels: matchLabels,
+			},
+		},
+	}, nil
+}
+
+// doesIngressMatch checks if application ingress in PublishingStrategy CR matches with IngressController CR
+func isOnCluster(publishingStrategyIngress *cloudingressv1alpha1.ApplicationIngress, ingressController *operatorv1.IngressController) bool {
+	// check to see if these fields are empty to ensure no nil pointer error
+	if string(ingressController.Status.EndpointPublishingStrategy.LoadBalancer.Scope) == "" || ingressController.Spec.Domain == "" || ingressController.Spec.DefaultCertificate.Name == "" || ingressController.Namespace == "" {
+		return false
+	}
+
+	if string(publishingStrategyIngress.Listening) != string(ingressController.Status.EndpointPublishingStrategy.LoadBalancer.Scope) {
+		return false
+	}
+	if publishingStrategyIngress.DNSName != ingressController.Spec.Domain {
+		return false
+	}
+	if publishingStrategyIngress.Certificate.Name != ingressController.Spec.DefaultCertificate.Name {
+		return false
+	}
+	if publishingStrategyIngress.Certificate.Namespace != ingressController.Namespace {
```

Comment:
Will this ever match? My understanding is `ingressController.Namespace` will always be = `openshift-ingress-operator`

Reply:
agreed. removing this condition